File: code/richediting/src/main/java/org/fbme/ide/richediting/adapters/fbnetwork/PortCell.java
package org.fbme.ide.richediting.adapters.fbnetwork;

import jetbrains.mps.nodeEditor.cells.EditorCell_Collection;

import java.awt.*;

public interface PortCell {
    Point getPosition();

    int getWidth();

    int getHeight();

    void paintTrace(Graphics2D g, int x, int y);

    void relayout();

    Point getPortCoordinate();

    Point getPortPosition();

    Rectangle getPortBounds();

    Rectangle getDNDBounds();

    EditorCell_Collection getRootCell();
}

File: code/richediting/src/main/kotlin/org/fbme/ide/richediting/adapters/fbnetwork/PortCell.kt
package org.fbme.ide.richediting.adapters.fbnetwork

import jetbrains.mps.nodeEditor.cells.EditorCell_Collection
import java.awt.Graphics2D
import java.awt.Point
import java.awt.Rectangle

interface PortCell {
    val position: Point
    val width: Int
    val height: Int
    fun paintTrace(g: Graphics2D, x: Int, y: Int)
    fun relayout()
    val portCoordinate: Point
    val portPosition: Point
    val portBounds: Rectangle
    val dNDBounds: Rectangle
    val rootCell: EditorCell_Collection
}
